### ISSUE TITLE
Improve idempodency in CloudDataTransferServiceCreateJobOperator

### ISSUE THREAD

**Issue by mik-laj** [apache/airflow#8285]
**Description**

We should improve idempodency in [CloudDataTransferServiceCreateJobOperator](https://github.com/apache/airflow/blob/42eef38/airflow/providers/google/cloud/operators/cloud_storage_transfer_service.py#L157). If a Job ID is passed and an AlreadyExist exception raises, then the task should be successful.

This issue is due to [the March 9 service update](https://cloud.google.com/storage-transfer/docs/release-notes#March_09_2020).

> FEATURE: Storage Transfer Service supports custom job IDs.

To complete this task you must complete the following steps:
- [ ] Update the operator
- [ ] Update the example DAG
- [ ] Update the documentation
- [ ] Run system tests (final check)

If you haven't used the GCP yet, after creating the account you will [get $300](https://cloud.google.com/free), which will allow you to get to know these services better.

The implementation of this task will allow a better understanding of GCP services, as well as learn methods of testing that is required by the community. If anyone is interested in this task, I am willing to provide all the necessary tips and information.

**Use case / motivation**

N/A

**Related Issues**

https://github.com/apache/airflow/issues/8284

**Comment by khyurri** [comment]
I can get it, 
Looks like easy and helpful

**Comment by khyurri** [comment]
Still in progress. 

Right now i've got some troubles reproducing error:

1. Failed to obtain the location of the source S3 bucket. Additional details: Access Denied. 
But S3 credentials are valid.

2. Not clear how to setup GCP. I've created service account add added owner permission to bucket, but still have error 

```
Failed to obtain the location of the Google Cloud Storage (GCS) bucket gcp-transfer-first-target_2 due to insufficient permissions.  Please verify that the necessary permissions have been granted.
```

3. `aws_hook = AwsBaseHook(self.aws_conn_id)` in `TransferJobPreprocessor ` always raises `AirflowException('Either client_type or resource_type must be provided.')`
I've hardcoded this line: `aws_hook = AwsBaseHook(self.aws_conn_id, client_type="s3")`
